runtime/vm/classsupport.c

```diff
@@ -953,6 +953,12 @@ loadNonArrayClass(J9VMThread* vmThread, J9Module *j9module, U_8* className, UDAT
 
 	foundClass = hashClassTableAt(classLoader, className, classNameLength);
 	if (NULL != foundClass) {
+		if (IS_WARM_RUN(vmThread->javaVM)
+			&& classLoader != vmThread->javaVM->systemClassLoader
```

Comment:
`classLoader != vmThread->javaVM->systemClassLoader` why is this being done?

Reply:
it was sort of a bad hack. I noticed in jclcinit, they allocate object refs after calling loadNonArrayClass (findClass). Object ref resolution is only needed here for non-bootstrap classes was just a conclusion I came upon but not sure

Reply:
Some bootstrap classes are loaded very early (about ~70) before we can allocate classObjects for them. So the class object allocation has to be done at a later time. However, this is not true for all classes loaded by the bootstrap classloader.

Reply:
you can use instead `J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_CLASS_OBJECT_ASSIGNED)` to figure out if we are past the point where all classes should have objects allocated.